include/Stack.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include "Vector.h"
+
+class Stack {
+public:
+    Stack() = default;
+    Stack(const size_t& initial_size, const int& initial_value);
```

Comment:
A major part of code readability and maintability is its uniformity. 
In `Vector.h` you implemented: ```Vector(size_t initial_size, int initial_value);```
But here you added `const ref` to the params. 
You should choose one way of passing params and stick with it for the sake of uniformity.
Also, for trivial types like `int` it isn't necessary to pass as const ref, the compiler will probably omit that anyway and pass by value